mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -0,0 +1,188 @@
+          "bucketing_version");
+
+  private final Configuration conf;
+  private Table icebergTable = null;
+  private Properties catalogProperties;
+  private boolean deleteIcebergTable;
+  private FileIO deleteIo;
+  private TableMetadata deleteMetadata;
+
+  public HiveIcebergMetaHook(Configuration conf) {
+    this.conf = conf;
+  }
+
+  @Override
+  public void preCreateTable(org.apache.hadoop.hive.metastore.api.Table hmsTable) {
+    this.catalogProperties = getCatalogProperties(hmsTable);
+
+    // Set the table type even for non HiveCatalog based tables
+    hmsTable.getParameters().put(BaseMetastoreTableOperations.TABLE_TYPE_PROP,
+        BaseMetastoreTableOperations.ICEBERG_TABLE_TYPE_VALUE.toUpperCase());
+
+    if (!Catalogs.hiveCatalog(conf)) {
+      // If not using HiveCatalog check for existing table
+      try {
+        this.icebergTable = Catalogs.loadTable(conf, catalogProperties);
+
+        Preconditions.checkArgument(catalogProperties.getProperty(InputFormatConfig.TABLE_SCHEMA) == null,
+            "Iceberg table already created - can not use provided schema");
+        Preconditions.checkArgument(catalogProperties.getProperty(InputFormatConfig.PARTITION_SPEC) == null,
+            "Iceberg table already created - can not use provided partition specification");
+
+        LOG.info("Iceberg table already exists {}", icebergTable);
+
+        return;
+      } catch (NoSuchTableException nte) {
+        // If the table does not exist we will create it below
+      }
+    }
+
+    // If the table does not exist collect data for table creation
+    String schemaString = catalogProperties.getProperty(InputFormatConfig.TABLE_SCHEMA);
+    Preconditions.checkNotNull(schemaString, "Please provide a table schema");
+    // Just check if it is parsable, and later use for partition specification parsing
+    Schema schema = SchemaParser.fromJson(schemaString);
+
+    String specString = catalogProperties.getProperty(InputFormatConfig.PARTITION_SPEC);
+    if (specString != null) {
+      // Just check if it is parsable
+      PartitionSpecParser.fromJson(schema, specString);
+    }
+
+    // Allow purging table data if the table is created now and not set otherwise
+    if (hmsTable.getParameters().get(InputFormatConfig.EXTERNAL_TABLE_PURGE) == null) {
+      hmsTable.getParameters().put(InputFormatConfig.EXTERNAL_TABLE_PURGE, "TRUE");
+    }
+
+    // Remove creation related properties
+    PARAMETERS_TO_REMOVE.forEach(hmsTable.getParameters()::remove);
+  }
+
+  @Override
+  public void rollbackCreateTable(org.apache.hadoop.hive.metastore.api.Table hmsTable) {
+    // do nothing
+  }
+
+  @Override
+  public void commitCreateTable(org.apache.hadoop.hive.metastore.api.Table hmsTable) {
+    if (icebergTable == null) {
+      if (Catalogs.hiveCatalog(conf)) {
+        catalogProperties.put(TableProperties.ENGINE_HIVE_ENABLED, true);
```

Comment:
@pvary @rdblue @massdosage 
In the iceberg documentation i could see that
"To enable Hive support globally for an application, set iceberg.engine.hive.enabled=true in its Hadoop configuration."

The value of iceberg.engine.hive.enabled must be true in order to enable hive support. However, I could still use Iceberg's hive support even if I set iceberg.engine.hive.enabled to false. If i understand correctly, the value "iceberg.engine.hive.enabled" is irrelevant.

Because, if it is a hive catalog we are setting engine.hive.enabled as true. 
In https://github.com/apache/iceberg/blob/01bc864b8eb8c4ca4240af85f00f4e67c78c466e/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java#L486

We are initially evaluating the 'engine.hive.enabled' value. In cases where it's a Hive catalog, this value is consistently 'True', rendering the 'iceberg.engine.hive.enabled' as unnecessary.

Let me know, if i missed anything in my understanding

Reply:
@shivjha30: The code line you have highlighted is for overriding the global settings on table level. So you have 2 levels to enable the hive engine:
- Catalog level (ConfigProperties.ENGINE_HIVE_ENABLED) - you can set the config in the hadoop configuration of the HiveCatalog
- Table level (TableProperties.ENGINE_HIVE_ENABLED) - you can set the config in the table properties

Reply:
Thanks @pvary for getting back. 
I have comprehended the functionality of the two levels concerning the Hive engine activation. If I'm not mistaken, even when we set ConfigProperties.ENGINE_HIVE_ENABLED to FALSE in Hadoop configuration, we are passing the ConfigProperties.ENGINE_HIVE_ENABLED as TRUE in the HiveIcebergMetaHook#commitCreateTable if it's of HiveCatalog type. This action renders the table level configuration irrelevant.

Reply:
You are using the metahook if you are using Hive to create the table. We expect that you want to be able to read the table from Hive in this case, so you need the storage handlers to be set.

8n this case, you need the storage handlers on the classpath of the other readers as well.